src/bunit.web/TestDoubles/Authorization/TestAuthorizationContext.cs

```diff
@@ -85,6 +86,8 @@ public TestAuthorizationContext SetRoles(params string[] roles)
 			Roles = roles ?? Array.Empty<string>();
 			_authProvider.TriggerAuthenticationStateChanged(UserName, Roles);
 
+			_authService.SetRoles(Roles);
+
```

Comment:
```suggestion
			_authService.SetRoles(Roles);
			_authProvider.TriggerAuthenticationStateChanged(UserName, Roles);
```
Should these two lines not be in this order? I.e. first set roles in auth service and then notify?

Reply:
These are setting two separate, independent services, so order isn't important as long as both happen before method returns. But swapped calls if that makes it easier to understand.